gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -56,10 +56,13 @@ public void write(Bundle bundle, File rootFolder, Bundle rawBundle) {
         services.values().parallelStream().forEach(serviceEntity -> {
             final String id = serviceEntity.getId();
             final String name = serviceEntity.getName();
-            final PolicyMetadata policyMetadata = writePolicy(bundle, policyFolder, serviceEntity.getParentFolder().getId(), name, id, serviceEntity.getPolicyXML());
-            final Set<Dependency> policyDependencies = getPolicyDependencies(id, rawBundle);
-            policyMetadata.setUsedEntities(policyDependencies);
-            policyMetadataMap.put(policyMetadata.getNameWithPath(), policyMetadata);
+            final Folder folder = bundle.getFolderTree().getFolderById(serviceEntity.getParentFolder().getId());
+            final Path policyPath = bundle.getFolderTree().getPath(folder);
+            writePolicy(policyFolder, policyPath, name, serviceEntity.getPolicyXML());
+            final PolicyMetadata policyMetadata = createPolicyMetadata(id, rawBundle, null);
+            final String path = PathUtils.unixPathEndingWithSeparator(policyPath);
```

Comment:
Keep the policy path as part of PolicyMetadata, and annotate it using @JsonIgnore. 

This way, you could make use of policyMetadata.getPath() as key.

Reply:
done